Python/ast.c

```diff
@@ -2053,15 +2068,21 @@ ast_for_comprehension(struct compiling *c, const node *n)
 
         if (NCH(n) == 2) {
             is_async = 1;
-            REQ(CHILD(n, 0), NAME);
-            assert(strcmp(STR(CHILD(n, 0)), "async") == 0);
+            REQ(CHILD(n, 0), ASYNC);
             sync_n = CHILD(n, 1);
         }
         else {
             sync_n = CHILD(n, 0);
         }
         REQ(sync_n, sync_comp_for);
 
+        /* Async comprehensions only allowed in Python 3.6 and greater */
+        if (is_async && c->c_feature_version < 6) {
+            ast_error(c, n,
+                    "Async comprehensions are only supported in Python 3.6 and greater");
```

Comment:
Indentation here and in several other places is a bit unusual, I would rather justify it after `(`.

Reply:
OK, fixed. My guess is that at some point in the past that function had had a name that was 2 characters shorter. :-)